addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/DefaultRequestHandler.java

```diff
@@ -0,0 +1,221 @@
+    private String getPrinterStatusMessage(String statusMsg) {
+        Map<String, String> smap = new HashMap<>();
+
+        smap.put("processing", "Printing");
+        smap.put("inPowerSave", "Power Save");
+        smap.put("ready", "Idle");
+        smap.put("closeDoorOrCover", "Door Open");
+        smap.put("inkSystemInitializing", "Loading Ink");
+
+        return smap.getOrDefault(statusMsg, statusMsg);
+    }
+
+    @Override
+    public void checkStatus() {
+        WebRequester status = new WebRequester(endpointStatus, config);
+        logger.debug("Checking {} status", thing.getUID().toString());
+        WebRequesterChecker result = status.process();
+        switch (result.status()) {
+            case WebRequester.REQ_STATUS_OK:
+                NodeList statuses = result.document().getDocumentElement().getElementsByTagName("psdyn:Status");
+
+                String printerStatus = "";
+                for (int i = 0; i < statuses.getLength(); i++) {
+                    Element currInk = (Element) statuses.item(i);
+                    if (currInk.getElementsByTagName("pscat:StatusCategory").item(0).getTextContent() != "genuineHP") {
+                        printerStatus = getPrinterStatusMessage(
+                                currInk.getElementsByTagName("pscat:StatusCategory").item(0).getTextContent());
+                    }
+                }
+
+                updateState(new ChannelUID(thing.getUID(), CGROUP_STATUS, CHANNEL_STATUS),
+                        new StringType(printerStatus));
+
+                logger.debug("Check {} status ok", thing.getUID().toString());
+                break;
+
+            case WebRequester.REQ_STATUS_ENDPOINTFAIL:
+                goneOffline(true);
+                break;
+
+            case WebRequester.REQ_STATUS_DOCEXCEPTION:
+                logger.debug("Check {} status failed", thing.getUID().toString());
+                break;
+        }
+    }
+
+    @Override
+    public void checkUsage() {
+        WebRequester status = new WebRequester(endpointUsage, config);
+        logger.debug("Checking {} usage", thing.getUID().toString());
+        WebRequesterChecker result = status.process();
+        switch (result.status()) {
+            case WebRequester.REQ_STATUS_OK:
+                // Get Ink Levels
+                NodeList consumableInk = result.document().getDocumentElement()
+                        .getElementsByTagName("pudyn:Consumable");
+
+                try {
+                    for (int i = 0; i < consumableInk.getLength(); i++) {
+                        Element currInk = (Element) consumableInk.item(i);
+
+                        String inkName = currInk.getElementsByTagName("dd:MarkerColor").item(0).getTextContent();
+
+                        String consumeType = currInk.getElementsByTagName("dd:ConsumableTypeEnum").item(0)
+                                .getTextContent();
+
+                        if (consumeType.equalsIgnoreCase("printhead")) {
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

Check all